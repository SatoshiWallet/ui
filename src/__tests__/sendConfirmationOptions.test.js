/* globals jest describe it expect */
/* eslint-disable flowtype/require-valid-file-annotation */

import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import SendConfirmationOptions from '../components/common/SendConfirmationOptions.js'

describe('SendConfirmation', () => {
  it('should render with standard props', () => {
    const renderer = new ShallowRenderer()

    const props = {
      changeMiningFee: jest.fn,
      sendMaxSpend: jest.fn,
      uniqueIdentifierModalActivated: jest.fn,
      sourceWallet: {},
      currencyCode: 'BTC',
      isEditable: true
    }
    const actual = renderer.render(<SendConfirmationOptions {...props} />)

    expect(actual).toMatchSnapshot()
  })

  it('not editable', () => {
    const renderer = new ShallowRenderer()

    const props = {
      changeMiningFee: jest.fn,
      sendMaxSpend: jest.fn,
      uniqueIdentifierModalActivated: jest.fn,
      sourceWallet: {},
      currencyCode: 'BTC',
      isEditable: false
    }
    const actual = renderer.render(<SendConfirmationOptions {...props} />)

    expect(actual).toMatchSnapshot()
  })

  it('XMR', () => {
    const renderer = new ShallowRenderer()

    const props = {
      changeMiningFee: jest.fn,
      sendMaxSpend: jest.fn,
      uniqueIdentifierModalActivated: jest.fn,
      sourceWallet: {},
      currencyCode: 'XMR',
      isEditable: true
    }
    const actual = renderer.render(<SendConfirmationOptions {...props} />)

    expect(actual).toMatchSnapshot()
  })

  it('XRP', () => {
    const renderer = new ShallowRenderer()

    const props = {
      changeMiningFee: jest.fn,
      sendMaxSpend: jest.fn,
      uniqueIdentifierModalActivated: jest.fn,
      sourceWallet: {},
      currencyCode: 'XRP',
      isEditable: true
    }
    const actual = renderer.render(<SendConfirmationOptions {...props} />)

    expect(actual).toMatchSnapshot()
  })
})
