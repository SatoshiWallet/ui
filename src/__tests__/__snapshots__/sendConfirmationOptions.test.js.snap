// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendConfirmation XMR 1`] = `
<View>
  <ForwardRef(enhanceContext-Context.Consumer(Menu))
    onBackdropPress={[Function]}
    onClose={[Function]}
    onOpen={[Function]}
    onSelect={[Function]}
    renderer={[Function]}
    rendererProps={Object {}}
  >
    <ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))
      customStyles={Object {}}
      disabled={false}
      style={
        Object {
          "paddingHorizontal": 11.236619718309859,
          "paddingVertical": 5.618309859154929,
        }
      }
    >
      <FormattedText
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "SourceSansPro-Bold",
            "fontSize": 25.28239436619718,
            "paddingHorizontal": 11.236619718309859,
          }
        }
      >
        ⋮
      </FormattedText>
    </ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))>
    <ForwardRef(enhanceContext-Context.Consumer(MenuOptions))
      customStyles={Object {}}
      optionsContainerStyle={
        Object {
          "width": 231.75528169014086,
        }
      }
    >
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="CHANGE_MINING_FEE"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Object {
                "color": "#4A5157",
                "fontSize": 25.28239436619718,
              }
            }
          >
            Change Mining Fee
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="HELP"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Object {
                "color": "#4A5157",
                "fontSize": 25.28239436619718,
              }
            }
          >
            Help
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
    </ForwardRef(enhanceContext-Context.Consumer(MenuOptions))>
  </ForwardRef(enhanceContext-Context.Consumer(Menu))>
</View>
`;

exports[`SendConfirmation XRP 1`] = `
<View>
  <ForwardRef(enhanceContext-Context.Consumer(Menu))
    onBackdropPress={[Function]}
    onClose={[Function]}
    onOpen={[Function]}
    onSelect={[Function]}
    renderer={[Function]}
    rendererProps={Object {}}
  >
    <ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))
      customStyles={Object {}}
      disabled={false}
      style={
        Object {
          "paddingHorizontal": 11.236619718309859,
          "paddingVertical": 5.618309859154929,
        }
      }
    >
      <FormattedText
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "SourceSansPro-Bold",
            "fontSize": 25.28239436619718,
            "paddingHorizontal": 11.236619718309859,
          }
        }
      >
        ⋮
      </FormattedText>
    </ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))>
    <ForwardRef(enhanceContext-Context.Consumer(MenuOptions))
      customStyles={Object {}}
      optionsContainerStyle={
        Object {
          "width": 231.75528169014086,
        }
      }
    >
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="CHANGE_MINING_FEE"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Object {
                "color": "#4A5157",
                "fontSize": 25.28239436619718,
              }
            }
          >
            Change Mining Fee
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="SEND_MAX"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Array [
                Object {
                  "color": "#4A5157",
                  "fontSize": 25.28239436619718,
                },
                Object {
                  "color": "#F1AA19",
                },
              ]
            }
          >
            Send Max Amount
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="ADD_UNIQUE_IDENTIFIER"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Array [
                Object {
                  "color": "#4A5157",
                  "fontSize": 25.28239436619718,
                },
              ]
            }
          >
            Add Destination Tag
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="HELP"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Object {
                "color": "#4A5157",
                "fontSize": 25.28239436619718,
              }
            }
          >
            Help
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
    </ForwardRef(enhanceContext-Context.Consumer(MenuOptions))>
  </ForwardRef(enhanceContext-Context.Consumer(Menu))>
</View>
`;

exports[`SendConfirmation not editable 1`] = `
<View>
  <ForwardRef(enhanceContext-Context.Consumer(Menu))
    onBackdropPress={[Function]}
    onClose={[Function]}
    onOpen={[Function]}
    onSelect={[Function]}
    renderer={[Function]}
    rendererProps={Object {}}
  >
    <ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))
      customStyles={Object {}}
      disabled={false}
      style={
        Object {
          "paddingHorizontal": 11.236619718309859,
          "paddingVertical": 5.618309859154929,
        }
      }
    >
      <FormattedText
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "SourceSansPro-Bold",
            "fontSize": 25.28239436619718,
            "paddingHorizontal": 11.236619718309859,
          }
        }
      >
        ⋮
      </FormattedText>
    </ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))>
    <ForwardRef(enhanceContext-Context.Consumer(MenuOptions))
      customStyles={Object {}}
      optionsContainerStyle={
        Object {
          "width": 231.75528169014086,
        }
      }
    >
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="HELP"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Object {
                "color": "#4A5157",
                "fontSize": 25.28239436619718,
              }
            }
          >
            Help
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
    </ForwardRef(enhanceContext-Context.Consumer(MenuOptions))>
  </ForwardRef(enhanceContext-Context.Consumer(Menu))>
</View>
`;

exports[`SendConfirmation should render with standard props 1`] = `
<View>
  <ForwardRef(enhanceContext-Context.Consumer(Menu))
    onBackdropPress={[Function]}
    onClose={[Function]}
    onOpen={[Function]}
    onSelect={[Function]}
    renderer={[Function]}
    rendererProps={Object {}}
  >
    <ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))
      customStyles={Object {}}
      disabled={false}
      style={
        Object {
          "paddingHorizontal": 11.236619718309859,
          "paddingVertical": 5.618309859154929,
        }
      }
    >
      <FormattedText
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "SourceSansPro-Bold",
            "fontSize": 25.28239436619718,
            "paddingHorizontal": 11.236619718309859,
          }
        }
      >
        ⋮
      </FormattedText>
    </ForwardRef(enhanceContext-Context.Consumer(MenuTrigger))>
    <ForwardRef(enhanceContext-Context.Consumer(MenuOptions))
      customStyles={Object {}}
      optionsContainerStyle={
        Object {
          "width": 231.75528169014086,
        }
      }
    >
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="CHANGE_MINING_FEE"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Object {
                "color": "#4A5157",
                "fontSize": 25.28239436619718,
              }
            }
          >
            Change Mining Fee
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="SEND_MAX"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Array [
                Object {
                  "color": "#4A5157",
                  "fontSize": 25.28239436619718,
                },
                Object {
                  "color": "#F1AA19",
                },
              ]
            }
          >
            Send Max Amount
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
      <ForwardRef(enhanceContext-Context.Consumer(MenuOption))
        customStyles={Object {}}
        disableTouchable={false}
        disabled={false}
        style={
          Object {
            "borderBottomColor": "#D9E3ED",
            "borderBottomWidth": 1,
            "justifyContent": "center",
          }
        }
        value="HELP"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 8.427464788732394,
              "paddingVertical": 5.618309859154929,
            }
          }
        >
          <FormattedText
            style={
              Object {
                "color": "#4A5157",
                "fontSize": 25.28239436619718,
              }
            }
          >
            Help
          </FormattedText>
        </View>
      </ForwardRef(enhanceContext-Context.Consumer(MenuOption))>
    </ForwardRef(enhanceContext-Context.Consumer(MenuOptions))>
  </ForwardRef(enhanceContext-Context.Consumer(Menu))>
</View>
`;
